api/environments/dynamodb/wrappers/identity_wrapper.py

```diff
@@ -89,37 +91,49 @@ def get_all_items(
         environment_api_key: str,
         limit: int,
         start_key: dict[str, "TableAttributeValueTypeDef"] | None = None,
-        projection_expression: str = None,
+        projection_expression: str | None = None,
+        return_consumed_capacity: bool = False,
     ) -> "QueryOutputTableTypeDef":
         filter_expression = Key("environment_api_key").eq(environment_api_key)
         query_kwargs: "QueryInputRequestTypeDef" = {
             "IndexName": "environment_api_key-identifier-index",
             "KeyConditionExpression": filter_expression,
             "Limit": limit,
         }
-        if projection_expression:
-            query_kwargs["ProjectionExpression"] = projection_expression
-
         if start_key:
             query_kwargs["ExclusiveStartKey"] = start_key
+        if projection_expression:
+            query_kwargs["ProjectionExpression"] = projection_expression
+        if return_consumed_capacity:
+            query_kwargs["ReturnConsumedCapacity"] = "TOTAL"
```

Comment:
This seems a bit cryptic, I'd love a comment here.

Reply:
Added.